### ISSUE TITLE
Basic features

### ISSUE THREAD

**Issue by jayrulez** [Tsimiii/Bamboo#2]
Let's define the basic features here:

Starting with registration:

What are the fields required for registration?

First name
Last name
Email address
Password

**Comment by Tsimiii** [comment]
Yes, these are the mandatory fields. Maybe we can include some optional fields (like Facebook): birth date and gender, but it's not necessary.

**Comment by jayrulez** [comment]
Which database are you using?

I noticed your config is using localdb. Do you mind switching to mssql server?

**Comment by Tsimiii** [comment]
Yes, I used localdb because I just tested the technology. I was planning to use mssql server, so it will be perfect.

**Comment by jayrulez** [comment]
Switched to mssql on my end. I'll soon open a pull request.

Do you have experience with oauth/openidconnect or any sort of token based authentication?

**Comment by jayrulez** [comment]
@Tsimiii

**Comment by Tsimiii** [comment]
No, not really but tomorrow (it's 23.19 pm here) I will check some tutorials to catch up.

**Comment by jayrulez** [comment]
Ok, you can have a look at openiddict or identityserver4

**Comment by Tsimiii** [comment]
Ty, I started watching openiddict, haven't finished yet, I will continue tomorrow. But we don't need it for registration, just for login, do we?

**Comment by jayrulez** [comment]
Right. We actually don't need it at all but it's just convenient to use.

However, since you want to learn this stuff then we can implement our own login system.

**Comment by jayrulez** [comment]
@Tsimiii  How are you doing with the angular side of things?

Do you need help getting started with that?

**Comment by Tsimiii** [comment]
Based in https://angular.io/docs/ts/latest/quickstart.html tutorial, angular 2 worked for me in a full empty project, after npm run command.

In this project I tried to do the same thing but I have a problem: after npm run the website runs on port 3000, so it doesn't load anything; with ctrl+f5 in Visual Studio the website runs on port 57455, it loads everything, for example the /registration page, where I have included an angular component but it doesn't load this way. Could you help me please? I commited the angular 2 configs based on the tutorial, in app folder there is my angular component, and I tried to paste it in Registration.cshtml, just for testing.

**Comment by jayrulez** [comment]
Ok. I'll have a look at this later.

I think what needs to happen is for a task to be run while compiling the .net project to also compile the angular app and publish it to the wwwroot folder so it can be served as static files.

An alternative approach to to put the angular app in a separate project and serve it from node.

I'll let you know what's up as soon as I look at it.
